modules/gitops-argo-cd-round-robin-sharding-algorithm.adoc

```diff
@@ -0,0 +1,138 @@
+* Reduce the risk of bottlenecks
+* Improve overall performance and reliability of the Argo CD system
+
+The introduction of alternative sharding algorithms allows for further customization based on specific use cases. You can select the algorithm that best aligns with your deployment needs, which results in greater flexibility and adaptability in diverse operational scenarios.
+
+.Prerequisites
+* You have installed the {gitops-title} Operator in your cluster.
+* You have access to the {OCP} web console.
+* You have access to the cluster with `cluster-admin` privileges.
+
+.Procedure
+
+You can enable the `round-robin` sharding algorithm by either using the {OCP} web console or the command-line interface.
+
+* To enable the `round-robin` algorithm using the {OCP} web console, complete the following steps:
+
+. In the *Administrator* perspective of the web console, go to *Operators* → *Installed Operators*.
+
+. Click *{gitops-title}* from the installed operators and go to the *Argo CD* tab.
+
+. Click the Argo CD instance where you want to enable the `round-robin` sharding algorithm, for example `openshift-gitops`.
+
+. Click the *YAML* tab and edit the yaml file:
++
+.Example Argo CD instance with round-robin sharding algorithm enabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: openshift-gitops
+  namespace: openshift-gitops
+spec:
+  controller:
+    processors: {}
+    sharding:
+      enabled: true <1>
+      replicas: 3 <2>
+    env: <3>
+      - name: ARGOCD_CONTROLLER_SHARDING_ALGORITHM 
+        value: round-robin
+----
+<1> Set the `sharding.enabled` parameter to `true` to enable sharding.
+<2> Set the number of replicas to the desired value, for example `3`.
+<3> Set the sharding algorithm to `round-robin`.
+
+. Click *Save* to save the changes.
+
+* To enable the `round-robin` algorithm using the command-line interface, complete the following steps:
+
+. Edit the Argo CD instance YAML file to enable sharding:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"sharding":{"enabled":true}}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Configure the sharding algorithm to `round-robin`:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"env":[{"name":"ARGOCD_CONTROLLER_SHARDING_ALGORITHM","value":"round-robin"}]}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+. Set the number of replicas to the desired value:
++
+[source,terminal]
+----
+oc patch argocd <argocd_instance> -n <namespace> --patch='{"spec":{"controller":{"sharding":{"replicas":<value>}}}}' --type=merge
+----
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/<argocd_instance> patched
+----
+
+.Verification
```

Comment:
@ekristova I see this statement in the SME draft: "The command returns the name of each cluster and to which shard it is assigned. You should see that the cluster distribution across shards is even. Which means that if the number of clusters C is a multiple of the number of shards replicas R, then each shard must have the same number of assigned clusters N, i.e. C divided by R." There is also a CLI command and output.

Suggest discussing with SME the possibility of adding this info in downstream docs with the multicluster setup and related UI Steps, CLI command, and example outputs for command and verification, both in the UI and CLI.

Reply:
I am providing the statement from SME: "There is no direct `oc`  command that allows this, however, by watching the logs, one can see also to which shard each cluster is attached. But, that’s mixed with other logs of course.
You can grep the results with :  `oc logs <pod> | grep "processed by shard"`
One has to be in debug mode: `Cluster with id=%s will be processed by shard %d", id "`

Reply:
@ekristova Thanks for the input. If this is downstream relevant, add this information as a TIP.

Reply:
Hello @Srivaralakshmi as I actually find this part to be essential for the verification (the user can verify that the algorithm works as intended) I included it as a proper verification step :https://github.com/openshift/openshift-docs/pull/69199/files#diff-636c8b4f7115acb0e240104942cbd73a3c167cc39f76388f36795abf088443a2R200-R235
I added the similar for the console verification as well. Please let me know what you think, thank you!

Reply:
@ekristova LGTM